src/modules/rppt_tensor_augmentations.cpp

```diff
@@ -180,3 +182,287 @@ rppt_brightness_host(RppPtr_t srcPtr,
 
     return RPP_SUCCESS;
 }
+
+/******************** gamma_correction ********************/
+
+RppStatus
+rppt_gamma_correction_gpu(RppPtr_t srcPtr,
+                          RpptDescPtr srcDescPtr,
+                          RppPtr_t dstPtr,
+                          RpptDescPtr dstDescPtr,
+                          Rpp32f *gammaTensor,
+                          RpptROIPtr roiTensorPtrSrc,
+                          RpptRoiType roiType,
+                          rppHandle_t rppHandle)
+{
+#ifdef OCL_COMPILE
+
+#elif defined (HIP_COMPILE)
+
+    Rpp32u paramIndex = 0;
+    copy_param_float(gammaTensor, rpp::deref(rppHandle), paramIndex++);
+
+    if (srcDescPtr->dataType == RpptDataType::U8)
+    {
+        if (dstDescPtr->dataType == RpptDataType::U8)
+        {
+            gamma_correction_hip_tensor(static_cast<Rpp8u*>(srcPtr) + srcDescPtr->offset,
+                                        srcDescPtr,
+                                        static_cast<Rpp8u*>(dstPtr) + dstDescPtr->offset,
+                                        dstDescPtr,
+                                        roiTensorPtrSrc,
+                                        roiType,
+                                        rpp::deref(rppHandle));
+        }
+    }
+    else if (srcDescPtr->dataType == RpptDataType::F16)
+    {
+        if (dstDescPtr->dataType == RpptDataType::F16)
+        {
+            gamma_correction_hip_tensor(static_cast<half*>(srcPtr) + srcDescPtr->offset,
+                                        srcDescPtr,
+                                        static_cast<half*>(dstPtr) + dstDescPtr->offset,
+                                        dstDescPtr,
+                                        roiTensorPtrSrc,
+                                        roiType,
+                                        rpp::deref(rppHandle));
+        }
+    }
+    else if (srcDescPtr->dataType == RpptDataType::F32)
+    {
+        if (dstDescPtr->dataType == RpptDataType::F32)
+        {
+            gamma_correction_hip_tensor(static_cast<Rpp32f*>(srcPtr) + srcDescPtr->offset,
```

Comment:
I think offset value is in bytes. So adding to float * will be wrong

Reply:
Since we have 4 bit depths, RPP assumes that the offset from the user is number of pixels they want to move the image by. Please let me know if offset needs to be in bytes and I'll change the structure variable to srcDescPtr->offsetInBytes too.